docs/source/queries/result.md

```diff
@@ -19,91 +19,48 @@ return a `QueryResult` with rows represented as `Option<Vec<Row>>`.
 
-### Basic representation
-`Row` is a basic representation of a received row. It can be used by itself, but it's a bit awkward to use:
-```rust
-# extern crate scylla;
-# use scylla::Session;
-# use std::error::Error;
-# async fn check_only_compiles(session: &Session) -> Result<(), Box<dyn Error>> {
-if let Some(rows) = session.query_unpaged("SELECT a from ks.tab", &[]).await?.rows {
-    for row in rows {
-        let int_value: i32 = row.columns[0].as_ref().unwrap().as_int().unwrap();
-    }
-}
-# Ok(())
-# }
-```
-
-### Parsing using `into_typed`
-The driver provides a way to parse a row as a tuple of Rust types:
-```rust
-# extern crate scylla;
-# use scylla::Session;
-# use std::error::Error;
-# async fn check_only_compiles(session: &Session) -> Result<(), Box<dyn Error>> {
-use scylla::IntoTypedRows;
-
-// Parse row as a single column containing an int value
-if let Some(rows) = session.query_unpaged("SELECT a from ks.tab", &[]).await?.rows {
-    for row in rows {
-        let (int_value,): (i32,) = row.into_typed::<(i32,)>()?;
-    }
-}
-
-// rows.into_typed() converts a Vec of Rows to an iterator of parsing results
-if let Some(rows) = session.query_unpaged("SELECT a from ks.tab", &[]).await?.rows {
-    for row in rows.into_typed::<(i32,)>() {
-        let (int_value,): (i32,) = row?;
-    }
-}
-
-// Parse row as two columns containing an int and text columns
-if let Some(rows) = session.query_unpaged("SELECT a, b from ks.tab", &[]).await?.rows {
-    for row in rows.into_typed::<(i32, String)>() {
-        let (int_value, text_value): (i32, String) = row?;
-    }
-}
-# Ok(())
-# }
-```
-
 ## Parsing using convenience methods
-[`QueryResult`](https://docs.rs/scylla/latest/scylla/transport/query_result/struct.QueryResult.html) provides convenience methods for parsing rows.
+
+[`QueryRowsResult`](https://docs.rs/scylla/latest/scylla/transport/query_result/struct.QueryRowsResult.html) provides convenience methods for parsing rows.
 Here are a few of them:
-* `rows_typed::<RowT>()` - returns the rows parsed as the given type
-* `maybe_first_row_typed::<RowT>` - returns `Option<RowT>` containing first row from the result
-* `first_row_typed::<RowT>` - same as `maybe_first_row`, but fails without the first row
-* `single_row_typed::<RowT>` - same as `first_row`, but fails when there is more than one row
-* `result_not_rows()` - ensures that query response was not `rows`, helps avoid bugs
+* `rows::<RowT>()` - returns the rows parsed as the given type
+* `maybe_first_row_typed::<RowT>` - returns the first received row or `None` if there are no rows
+* `first_row::<RowT>` - returns the first received row; fails if there are no rows
+* `single_row::<RowT>` - same as `first_row`, but fails when there is more than one row
 
+Additionally, [`QueryResult`](https://docs.rs/scylla/latest/scylla/transport/query_result/struct.QueryResult.html) has a method `result_not_rows()`, which ensures that query response was not `rows` and thus helps avoid bugs.
```

Comment:
Maybe we should mention how to obtain `QueryRowsResult` from `QueryResult` as it's not exactly clear after reading this section (without looking at examples).